### PR TITLE
DROID-2275 Payment |  Api

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt [hunk 1]
```diff
@@ -1986,4 +1986,121 @@ class MiddlewareServiceImplementation @Inject constructor(
...
+        val error = response.error
+        if (error != null && error.code != Rpc.Membership.Finalize.Response.Error.Code.NULL) {
+            throw Exception(error.description)
+        } else {
+            return response
+        }
+    }
+
+    override fun getVerificationEmailStatus(request: Rpc.Membership.GetVerificationEmailStatus.Request): Rpc.Membership.GetVerificationEmailStatus.Response {
+        val encoded = Service.membershipGetVerificationEmail(
```
Comment (uburoiubu) [inline middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt:2056]: membershipGetVerificationEmailStatus?

File: middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt [hunk 2]
```diff
@@ -1986,4 +1986,121 @@ class MiddlewareServiceImplementation @Inject constructor(
...
+        val response = Rpc.Membership.GetPortalLinkUrl.Response.ADAPTER.decode(encoded)
+        val error = response.error
+        if (error != null && error.code != Rpc.Membership.GetPortalLinkUrl.Response.Error.Code.NULL) {
+            throw Exception(error.description)
+        } else {
+            return response
+        }
+    }
+
+    override fun finalizeMembership(request: Rpc.Membership.Finalize.Request): Rpc.Membership.Finalize.Response {
```
Comment (uburoiubu) [inline middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt:2042]: membershipFinalize

File: middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt [hunk 3]
```diff
@@ -1986,4 +1986,121 @@ class MiddlewareServiceImplementation @Inject constructor(
...
+        val response = Rpc.Membership.GetPaymentUrl.Response.ADAPTER.decode(encoded)
+        val error = response.error
+        if (error != null && error.code != Rpc.Membership.GetPaymentUrl.Response.Error.Code.NULL) {
+            throw Exception(error.description)
+        } else {
+            return response
+        }
+    }
+
+    override fun getPortalLinkUrl(request: Rpc.Membership.GetPortalLinkUrl.Request): Rpc.Membership.GetPortalLinkUrl.Response {
```
Comment (uburoiubu) [inline middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt:2029]: membershipGetPortalLinkUrl

File: middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt [hunk 4]
```diff
@@ -1986,4 +1986,121 @@ class MiddlewareServiceImplementation @Inject constructor(
...
+        val response = Rpc.Membership.IsNameValid.Response.ADAPTER.decode(encoded)
+        val error = response.error
+        if (error != null && error.code != Rpc.Membership.IsNameValid.Response.Error.Code.NULL) {
+            throw Exception(error.description)
+        } else {
+            return response
+        }
+    }
+
+    override fun getPaymentUrl(request: Rpc.Membership.GetPaymentUrl.Request): Rpc.Membership.GetPaymentUrl.Response {
```
Comment (uburoiubu) [inline middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt:2016]: membershipGetPaymentUrl

File: middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt [hunk 5]
```diff
@@ -1986,4 +1986,121 @@ class MiddlewareServiceImplementation @Inject constructor(
...
+        val response = Rpc.Membership.GetStatus.Response.ADAPTER.decode(encoded)
+        val error = response.error
+        if (error != null && error.code != Rpc.Membership.GetStatus.Response.Error.Code.NULL) {
+            throw Exception(error.description)
+        } else {
+            return response
+        }
+    }
+
+    override fun isNameValid(request: Rpc.Membership.IsNameValid.Request): Rpc.Membership.IsNameValid.Response {
```
Comment (uburoiubu) [inline middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt:2003]: membershipIsNameValid

**Review by uburoiubu — APPROVED** [review]

File: middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt [hunk 6]
```diff
@@ -1986,4 +1986,121 @@ class MiddlewareServiceImplementation @Inject constructor(
             return response
         }
     }
+
+    override fun getStatus(request: Rpc.Membership.GetStatus.Request): Rpc.Membership.GetStatus.Response {
```
[the remark this answers is not in the document]
Comment (konstantiniiv) [reply middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt]: done

File: middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt (hunk 2 above)
Comment (konstantiniiv) [reply middleware/src/main/java/com/anytypeio/anytype/middleware/service/MiddlewareServiceImplementation.kt]: done

**Pull request merged**